numpy/distutils/command/build_ext.py

```diff
@@ -262,15 +262,24 @@ def run(self):
         self.build_extensions()
 
         # Copy over any extra DLL files
-        runtime_lib_dir = os.path.join(
-            self.build_lib, self.distribution.get_name(), 'extra-dll')
-        for fn in os.listdir(self.extra_dll_dir):
-            if not fn.lower().endswith('.dll'):
-                continue
-            if not os.path.isdir(runtime_lib_dir):
-                os.makedirs(runtime_lib_dir)
-            runtime_lib = os.path.join(self.extra_dll_dir, fn)
-            copy_file(runtime_lib, runtime_lib_dir)
+        # FIXME: In the case where there are more than two packages,
+        # we blindly assume that both packages need all of the libraries,
+        # resulting in a larger wheel than is required. This should be fixed,
+        # but it's so rare that I won't bother to handle it.
+        pkg_roots = set(
+            self.get_ext_fullname(ext.name).split('.')[0]
```

Comment:
Does it do the right thing with namespace packages?

Reply:
I'm using pretty much [exactly the same logic](https://github.com/pypa/auditwheel/blob/5b8b569d84d21153877d16aaa6786ddd2e0a0411/auditwheel/repair.py#L55) as auditwheel. In the case of a namespace package, `__config__` **should** locate the `extra-dll-dir` as relative to the package root (though it doesn't currently nor with this PR). Is `package.subpackage.__config__` a thing?